tap_s3_csv/csv_iterator.py

```diff
@@ -41,4 +45,30 @@ def get_row_iterator(iterable, options=None):
             raise Exception('CSV file missing date_overrides headers: {}, file only contains headers for fields: {}'
                             .format(date_overrides - headers, headers))
 
+    reader.fieldnames = handle_empty_fieldnames(reader.fieldnames, fieldname_pool, options)
+
```

Comment:
csv.DictReader's __next__ method uses self.fieldnames to zip the row data and this data is passed in as argument for transform function to transform into record.
https://github.com/python/cpython/blob/8136606769661c103c46d142e52ecbbbb88803f6/Lib/csv.py#L108